packages/cli-repl/src/history.ts

```diff
@@ -0,0 +1,20 @@
+import redactInfo from 'mongodb-redact';
+
+/**
+ * Modifies command history array based on sensitive information.
+ * If redact option is passed, also redacts sensitive info.
+ *
+ * @param {array} History - Array of commands, where the first command is the
+ * most recent.
+ *
+ * @param {boolean} Redact - Option to redact sensitive info.
+ */
+function changeHistory(history: string[], redact: boolean = false) {
```

Comment:
Following up on what we discussed: what about exporting a function that takes a history entry and returns either a new history entry or falsy? ie.

``` js
function filterHistoryEntry(entry) {
  const hiddenCommands = RegExp('createUser|auth|updateUser|changeUserPassword', 'ig');

  if (hiddenCommands.test(entry)) return;
  return redactInfo(entry);
}
```

NOTE: If it is ok for you I can extract it after we merge and i rebase from master.

Reply:
i made this function exportable so you can just require it from cli-repl, so there is no rush to extract it if you want to use it right away.

for it to work in node's cli, i have to modify the history object and I think the cleanest way is just modifying then having a new history entry or falsy. This function will return what you passed it in if there are no modifications necessary. Does that make sense? There are examples in the test file of what I mean.

Reply:
Not sure I understood what you mean there :). 

I would not use that function as is now, I just need the core feature, as that would result just in plain duplicated code.

Why do you think is wrong to extract the function and using it like this:

``` js
import historyFilter from 'whatever shared place';

function changeHistory(history: string[], redact: boolean = false) {
   const filteredEntry = historyFilter(history[0], redact);
   if (!filteredEntry) {
      history.shift();
      return;
   }

   history[0] = filteredEntry;
}
```

Reply:
hmmmm so for me history is an array of strings, where the first item is the latest. The way I use this function now is quite clean:

```js
      repl.on('flushHistory', function() {
        changeHistory(repl.history, redactInfo);
      })
```
So using your proposed API is a lot less clean for my use case.

I also find it weird if when an API endpoint returns a false _or_ returns a value. Imo APIs should in generally return same structure, or to be more specific the same data type, of a thing, since it makes them more predictable to use.

Reply:
Ok not holding further on that opinion then, i'll work with that. Although I still disagree (especially on the sort order) :).

Reply:
yea maybe let's talk about history a bit more irl next week! Maybe I am completely misunderstanding this \o/

I think the func is so very smol, we can change it anytime if it turns out to be a bad idea™